app/javascript/components/CheckoutDashboard/DiscountsPage.tsx

```diff
@@ -379,7 +379,8 @@ const DiscountsPage = ({ offer_codes, pages, products, pagination: initialPagina
                         <div className="grid grid-cols-[min-content_1fr] gap-2">
                           {validAt && currentDate < validAt ? (
                             <>Scheduled</>
-                          ) : expiresAt && currentDate > expiresAt ? (
+                          ) : (expiresAt && currentDate > expiresAt) ||
+                            (statistics && offerCode.limit && statistics.uses.total >= offerCode.limit) ? (
```

Comment:
Can we reflect valid = false when uses.total is reached?

Reply:
We can have this ^ in the back-end and then the front-end is just displaying "Limit reached" if uses.total == limit.